sqlitecluster/SQLiteNode.cpp

```diff
@@ -2812,3 +2717,97 @@ string SQLiteNode::_getLostQuorumLogMessage() const {
+            } else if (_state == SQLiteNodeState::LEADING) {
+                PINFO("Higher-priority peer is trying to stand up while we are LEADING, STANDINGDOWN.");
+                _changeState(SQLiteNodeState::STANDINGDOWN);
+            } else {
+                PWARN("Higher-priority peer is trying to stand up while we are STANDINGDOWN, continuing.");
+            }
+        } else {
+            // Deny because we're currently in the process of leading and we're higher priority.
+            response["Response"] = "deny";
+            response["Reason"] = "I am leading";
+
+            // Hmm, why is a lower priority peer trying to stand up? Is it possible we're no longer in
+            // control of the cluster? Let's see how many nodes are subscribed.
+            if (_majoritySubscribed()) {
+                // we have a majority of the cluster, so ignore this oddity.
+                PHMMM("Lower-priority peer is trying to stand up while we are " << stateName(_state)
+                      << " with a majority of the cluster; denying and ignoring.");
+            } else {
+                // We don't have a majority of the cluster -- maybe it knows something we don't?  For
+                // example, it could be that the rest of the cluster has forked away from us. This can
+                // happen if the leader hangs while processing a command: by the time it finishes, the
+                // cluster might have elected a new leader, forked, and be a thousand commits in the future.
+                // In this case, let's just reset everything anyway to be safe.
+                PWARN("Lower-priority peer is trying to stand up while we are " << stateName(_state)
+                      << ", but we don't have a majority of the cluster so reconnecting and SEARCHING.");
+                _reconnectAll();
+                // TODO: This puts us in an ambiguous state if we switch to SEARCHING from LEADING,
+                // without going through the STANDDOWN process. We'll need to handle it better, but it's
+                // unclear if this can ever happen at all. exit() may be a reasonable strategy here.
+                _changeState(SQLiteNodeState::SEARCHING);
+            }
+        }
+    } else {
+        // Approve if nobody else is trying to stand up
+        response["Response"] = "approve"; // Optimistic; will override
+        for (auto otherPeer : _peerList) {
+            if (otherPeer != peer) {
+                // See if it's trying to be leader
+                if (otherPeer->state == SQLiteNodeState::STANDINGUP || otherPeer->state == SQLiteNodeState::LEADING || otherPeer->state == SQLiteNodeState::STANDINGDOWN) {
+                    // We need to contest this standup
+                    response["Response"] = "deny";
+                    response["Reason"] = "peer '" + otherPeer->name + "' is '" + stateName(otherPeer->state) + "'";
+                    break;
+                }
```

Comment:
I know this is not related to the changes you're applying, but just wanted to understand this part. In the code above, if a peer is trying to standup and we're standing down, we add a warn (line 2762) but don't really set anything in the response. But in this block, if we see another peer standing down, we actually deny it. 

Why do we have that different behavior between both?

Reply:
I think this is just an oversight where we've used a definition of leading in this case that includes standingup and standingdown. We could probably exclude it here as well and only use `deny` if the peer is standingup or leading.